var/spack/repos/builtin/packages/wgrib2/package.py

```diff
@@ -2,30 +2,65 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-import os
 import re
 
+import spack.build_systems.cmake
+import spack.build_systems.makefile
 from spack.package import *
 
-
-class Wgrib2(MakefilePackage):
+variant_map_common = {
+    "netcdf3": "USE_NETCDF3",
+    "netcdf4": "USE_NETCDF4",
+    "mysql": "USE_MYSQL",
+    "udf": "USE_UDF",
+    "regex": "USE_REGEX",
+    "tigge": "USE_TIGGE",
+    "proj4": "USE_PROJ4",
+    "aec": "USE_AEC",
+    "g2c": "USE_G2CLIB",
+    "png": "USE_PNG",
+    "jasper": "USE_JASPER",
+    "openmp": "USE_OPENMP",
+    "wmo_validation": "USE_WMO_VALIDATION",
+    "ipolates": "USE_IPOLATES",
+    "disable_alarm": "DISABLE_ALARM",
+    "fortran_api": "MAKE_FTN_API",
+    "disable_stat": "DISABLE_STAT",
+    "openjpeg": "USE_OPENJPEG",
+}
+
+variant_map_makefile = {"spectral": "USE_SPECTRAL"}
+
+variant_map_cmake = {
+    "netcdf": "USE_NETCDF",
+    "disable_timezone": "DISABLE_TIMEZONE",
+    "enable_docs": "ENABLE_DOCS",
+}
+
+variant_map_makefile.update(variant_map_common)
+variant_map_cmake.update(variant_map_common)
+
+
+class Wgrib2(MakefilePackage, CMakePackage):
     """Utility for interacting with GRIB2 files"""
 
     homepage = "https://www.cpc.ncep.noaa.gov/products/wesley/wgrib2"
     url = "https://www.ftp.cpc.ncep.noaa.gov/wd51we/wgrib2/wgrib2.tgz.v2.0.8"
+    git = "https://github.com/NOAA-EMC/wgrib2"
 
-    maintainers("t-brown", "AlexanderRichert-NOAA", "Hang-Lei-NOAA", "edwardhartnett")
-
-    version(
-        "3.1.1",
-        sha256="9236f6afddad76d868c2cfdf5c4227f5bdda5e85ae40c18bafb37218e49bc04a",
-        extension="tar.gz",
-    )
-    version(
-        "3.1.0",
-        sha256="5757ef9016b19ae87491918e0853dce2d3616b14f8c42efe3b2f41219c16b78f",
-        extension="tar.gz",
+    maintainers(
+        "AlysonStahl-NOAA", "t-brown", "AlexanderRichert-NOAA", "Hang-Lei-NOAA", "edwardhartnett"
     )
+
+    build_system(conditional("cmake", when="@3.2:"), conditional("makefile", when="@:3.1"))
```

Comment:
See https://spack.readthedocs.io/en/latest/packaging_guide.html#multiple-build-systems:

> Since the package supports multiple build systems, it is necessary to declare which one is the default.

```suggestion
    build_system(conditional("cmake", when="@3.2:"), conditional("makefile", when="@:3.1"), default="cmake")
```